WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -752,6 +755,14 @@ public void onAccountChanged(OnAccountChanged event) {
         // Sign-out is handled in `handleSiteRemoved`, no need to show the signup flow here
         if (mAccountStore.hasAccessToken()) {
             mBottomNav.showNoteBadge(mAccountStore.getAccount().getHasUnseenNotes());
+            if (AppPrefs.getShouldTrackMagicLinkSignup()) {
```

Comment:
As we discussed on the call, I think it'd be a good improvement to have this logic by itself in a helper method.

```
private void trackMagicLinkSignupIfNecessary() {
    if (AppPrefs.getShouldTrackMagicLinkSignup()) {
        AccountModel account = mAccountStore.getAccount();
        if (!TextUtils.isEmpty(account.getUserName()) && !TextUtils.isEmpty(account.getEmail())) {
            mLoginAnalyticsListener.trackCreatedAccount(account.getUserName(), account.getEmail());
            mLoginAnalyticsListener.trackSignupMagicLinkSucceeded();
            AppPrefs.removeShouldTrackMagicLinkSignup();
        }
    }
}
```

Reply:
Arg! Yes, this change slipped my mind.